doc/source/user/absolute_beginners.rst

```diff
@@ -0,0 +1,1913 @@
+on arrays would be extremely inefficient if the arrays weren't homogenous. 
+
+**Why use NumPy?**
+
+NumPy arrays are faster and more compact than Python lists. An array consumes
+less memory and is convenient to use. NumPy uses much less memory to store data
+and it provides a mechanism of specifying the data types. This allows the code
+to be optimized even further. 
+
+What is an array? 
+-----------------
+
+An array is a central data structure of the NumPy library. An array is a grid of
+values and it contains information about the raw data, how to locate an element,
+and how to interpret an element. It has a grid of elements that can be indexed
+in `various ways <https://numpy.org/devdocs/user/quickstart.html#indexing-slicing-and-iterating>`_.
+The elements are all of the same type, referred to as the array ``dtype``. 
+
+An array can be indexed by a tuple of nonnegative integers, by booleans, by
+another array, or by integers. The ``rank`` of the array is the number of
+dimensions. The ``shape`` of the array is a tuple of integers giving the size of
+the array along each dimension.
+
+One way we can initialize NumPy arrays is from Python lists, using nested lists
+for two- or higher-dimensional data.. 
+
+For example::
+
+  >>> array = np.array([1, 2, 3, 4, 5, 6])
+
+or::
+
+  >>> a = np.array([[1, 2, 3, 4], [5, 6, 7, 8], [9, 10, 11, 12]])
+
+We can access the elements in the array using square brackets. When you're
+accessing elements, remember that indexing in NumPy starts at 0. That means that
+if you want to access the first element in your array, you'll be accessing
+element "0".
+
+::
+
+  >>> print(a[0])
+  [1 2 3 4]
+
+
+More information about arrays 
+-----------------------------
+
+::
+
+  1D array, 2D array, ndarray, vector, matrix
+
+------
+
+You might occasionally hear an array referred to as a "ndarray," which is
+shorthand for "N-dimensional array." An N-dimensional array is simply an array
+with any number of dimensions. You might also hear **1-D**, or one-dimensional
+array, **2-D**, or two-dimensional array, and so on. The NumPy ``ndarray`` class
+is used to represent both matrices and vectors. A **vector** is an array with a
+single column, while a **matrix** refers to an array with multiple columns.
+
+**What are the attributes of an array?**
+
+An array is usually a fixed-size container of items of the same type and size.
+The number of dimensions and items in an array is defined by its shape. The
+shape of an array is a tuple of non-negative integers that specify the sizes of
+each dimension. 
+
+In NumPy, dimensions are called **axes**. This means that if you have a 2D array
+that looks like this::
+
+  [[0., 0., 0.],
+   [1., 1., 1.]]
+
+Your array has 2 axes. The first axis has a length of 2 and the second axis has
+a length of 3.
+
+Just like in other Python container objects, the contents of an array can be
+accessed and modified by indexing or slicing the array. Different arrays can
```

Comment:
People are often surprised that numpy arrays do _not_ behave just like other containers in the way this second sentence describes - probably worth qualifying somehow, will leave the best place to draw attention to this up to you.
```suggestion
accessed and modified by indexing or slicing the array. Unlike the typical container objects, different arrays can
```